components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/CorrelationConfigDAO.java

```diff
@@ -0,0 +1,280 @@
+ * under the License.
+ *
+ */
+
+package org.wso2.carbon.apimgt.impl.dao;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.apimgt.api.APIManagementException;
+import org.wso2.carbon.apimgt.api.ExceptionCodes;
+import org.wso2.carbon.apimgt.impl.APIConstants;
+import org.wso2.carbon.apimgt.impl.dao.constants.SQLConstants;
+import org.wso2.carbon.apimgt.impl.dto.CorrelationConfigDTO;
+import org.wso2.carbon.apimgt.impl.dto.CorrelationConfigPropertyDTO;
+import org.wso2.carbon.apimgt.impl.internal.ServiceReferenceHolder;
+import org.wso2.carbon.apimgt.impl.utils.APIMgtDBUtil;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Database Access Library for the Correlation Configs Feature.
+ */
+public class CorrelationConfigDAO {
+    private static final Log log = LogFactory.getLog(CorrelationConfigDAO.class);
+    private static final CorrelationConfigDAO correlationConfigDAO = new CorrelationConfigDAO();
+
+    private static final String COMPONENT_NAME = "COMPONENT_NAME";
+    private static final String ENABLED = "ENABLED";
+    private static final String PROPERTY_NAME = "PROPERTY_NAME";
+    private static final String PROPERTY_VALUE = "PROPERTY_VALUE";
+    private static final String DENIED_THREADS = "deniedThreads";
+    private List<String> correlationComponents;
+    private static final String DEFAULT_DENIED_THREADS = "MessageDeliveryTaskThreadPool,HumanTaskServer," +
+            "BPELServer,CarbonDeploymentSchedulerThread";
+
+    private CorrelationConfigDAO() {
+        correlationComponents = ServiceReferenceHolder.getInstance().getAPIManagerConfigurationService()
+                .getAPIManagerConfiguration().getProperty(APIConstants.CORRELATION_LOG_COMPONENTS);
+        correlationComponents.replaceAll(String::trim);
+        log.info("Correlation Components " + correlationComponents.toString());
+    }
+
+    public static CorrelationConfigDAO getInstance() {
+        return correlationConfigDAO;
+    }
+
+    public boolean updateCorrelationConfigs(List<CorrelationConfigDTO> correlationConfigDTOList)
+            throws APIManagementException {
+
+        for (CorrelationConfigDTO correlationConfigDTO : correlationConfigDTOList) {
+            String componentName = correlationConfigDTO.getName().trim();
+            if (!correlationComponents.contains(componentName)) {
+                throw new APIManagementException("Invalid Component Name : " + componentName,
+                        ExceptionCodes.from(ExceptionCodes.CORRELATION_CONFIG_BAD_REQUEST_INVALID_NAME));
+            }
+        }
+        String queryConfigs = SQLConstants.UPDATE_CORRELATION_CONFIGS;
+        String queryProps = SQLConstants.UPDATE_CORRELATION_CONFIG_PROPERTIES;
+        try (Connection connection = APIMgtDBUtil.getConnection()) {
+
```

Comment:
Remove empty line

Reply:
> Remove empty line

These changes were authored and committed by @pasindu-sandima. I will send a separate PR adding these requested changes since this PR is already merged.